### ISSUE TITLE
OverlappingFileLockException when headerRecovery results in resizeRafIfTooSmall

### ISSUE THREAD

**Issue by nicktindall** [OpenHFT/Chronicle-Queue#1291]
When `net.openhft.chronicle.queue.impl.single.SingleChronicleQueue.StoreSupplier#headerRecovery` executes and it results in a `net.openhft.chronicle.bytes.internal.ChunkedMappedFile#resizeRafIfTooSmall`, you end up attempting to lock the file channel twice. The lock is not re-entrant, so the OverlappingFileLockException is thrown. We might need to implement a re-entrant layer in ChunkedMappedFile or something.

Example stacktrace:
```
Exception in thread "main" java.nio.channels.OverlappingFileLockException
  at sun.nio.ch.SharedFileLockTable.checkList(FileLockTable.java:255)
  at sun.nio.ch.SharedFileLockTable.add(FileLockTable.java:152)
  at sun.nio.ch.FileChannelImpl.lock(FileChannelImpl.java:1062)
  at java.nio.channels.FileChannel.lock(FileChannel.java:1053)
  at net.openhft.chronicle.bytes.internal.ChunkedMappedFile.resizeRafIfTooSmall(ChunkedMappedFile.java:258)
  at net.openhft.chronicle.bytes.internal.ChunkedMappedFile.acquireByteStore(ChunkedMappedFile.java:181)
  at net.openhft.chronicle.bytes.MappedFile.acquireByteStore(MappedFile.java:197)
  at net.openhft.chronicle.bytes.internal.ChunkedMappedBytes.acquireNextByteStore0(ChunkedMappedBytes.java:338)
  at net.openhft.chronicle.bytes.internal.ChunkedMappedBytes.readVolatileInt(ChunkedMappedBytes.java:468)
  at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue$StoreSupplier.headerRecovery(SingleChronicleQueue.java:1089)
  at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue$StoreSupplier.acquire(SingleChronicleQueue.java:1043)
  at net.openhft.chronicle.queue.impl.WireStorePool.acquire(WireStorePool.java:56)
  at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue.storeForCycle(SingleChronicleQueue.java:574)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.cycle(StoreTailer.java:1128)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.nextIndexWithNextAvailableCycle0(StoreTailer.java:521)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.nextIndexWithNextAvailableCycle(StoreTailer.java:505)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.endOfCycle(StoreTailer.java:342)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.next0(StoreTailer.java:315)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument0(StoreTailer.java:237)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument(StoreTailer.java:212)
  at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument(StoreTailer.java:156)
  at software.chronicle.enterprise.queue.replication.stress.common.Consumer.consumeQueue(Consumer.java:107)
  at software.chronicle.enterprise.queue.replication.stress.common.Consumer.consumeQueues(Consumer.java:97)
  at software.chronicle.enterprise.queue.replication.stress.common.Consumer.run(Consumer.java:67)
  at software.chronicle.enterprise.queue.replication.stress.UngracefulShutdownTest$UngracefulShutdownConsumer.main(UngracefulShutdownTest.java:255)
```

**Comment by nicktindall** [comment]
This _may_ only occur when forcible=true, which is relatively new, we previously only tested the equivalent of forcible=false. (forcible means kill -9)

See https://teamcity.chronicle.software/test/-3874020191641721563?currentProjectId=Chronicle_BuildAll&expandTestHistoryChartSection=true

**Comment by nicktindall** [comment]
Confirmed this is due to the presence of an empty roll cycle at the end of the queue, probably resulting from the `kill -9`. Granted this is an error case, but we should be able to handle it more gracefully.

Reproduction/test implemented in 53889feba2dd6f12a9947c1266057ec2da3361c2

It has highlighted that we need to manage those file locks more carefully in the general case.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.23.42](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.23.42), [BOM-2.23.153](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.23.153)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.24ea8](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.24ea8), [BOM-2.24ea41](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.24ea41)
